### PR TITLE
Heap Sort Typescript Implementation

### PR DESCRIPTION
**Pull request by aayushbest**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [ ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [ ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [ ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [ ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [ ] This pull request shall only be merged once the build passes.

-   [ ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-hsrt/typescript/heap_sort.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+}
+
+// Driver program
+
+function main() {
+  const sample: Array<number> = [12, 11, 13, 5, 6, 7];
+  heapSort(sample);
+  console.log(sample);
```
Comment (RavinduSachintha) [inline algorithms/ar-hsrt/typescript/heap_sort.ts:61]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6375722)

**Comment by Danushka96** [comment]
Hi @aayushbest 
Can you update the Readme.md algorithm chart by adding the implemented language?
Update the readme and send it to this PR,
Thanks

**Comment by aayushbest** [comment]
Sure.

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
@all-contributors please add @aayushbest  for code and doc

**Comment by Danushka96** [comment]
Hi @aayushbest 
Thanks for your contribution,

**Comment by aayushbest** [comment]
Thankyou so much @Danushka96 . The pleasure is all mine. Happy to be a contributor for this project. Seeking for your approval for more contributions in multiple languages.

**Comment by Danushka96** [comment]
Great, 🙂

**Pull request merged**